experimental/framework/fluid-static/src/containerCode.ts

```diff
@@ -47,6 +48,57 @@ export interface FluidContainer
      * The returned object needs to be stored via handle by the caller.
      */
     create<T extends IFluidLoadable>(objectClass: LoadableObjectClass<T>): Promise<T>;
+
+    /**
+     * If the container is created in a detached state, this will connect it to the service. This allows the
+     * data in the container to be permanently persisted instead of being in-memory. This will throw
+     * an error if the container is already attached when this is called.
+     */
+    attachToService: (config: C) => Promise<void>;
```

Comment:
this binds the IFluidContainer to the serice client types for the config. i'd consider reversing this, and passing the FluidContianer to the client with the config to keep the FluidContainer type reusable across services

Reply:
I was attempting this but the issue I'm facing here is that the Client does not have access to the `Container` within `FluidContainer` since it is kept private to not expose it on the developer surface. The `FluidContainer` interface limits us to whatever the 3P developer would have access to but here we need to reach into the workings of the `Container` itself.

We also don't want to expose `IRequest` on the `FluidContainer` interface This would have allowed for something like `attachToService: (request: IRequest) => Promise<void>`. Each service could just pass its own generated request and let `FluidContainer` handle the job of actually attaching, i.e.

```
public async attachContainer(
        serviceContainerConfig: TinyliciousContainerConfig,
        fluidContainer: FluidContainer
    ): Promise<void> {
        const request = { url: serviceContainerConfig.id };
        fluidContainer.attachToService(request)
    }
```
But based on user feedback, we want to avoid using `IRequest` on the developer surface.

One way would be to access the FluidContainer's private runtime from TinyliciousClient but that seems super hacky:
```
  public async attachContainer(
      serviceContainerConfig: TinyliciousContainerConfig,
      fluidContainer: FluidContainer<TinyliciousContainerConfig>
  ): Promise<void> {
      const request = { url: serviceContainerConfig.id };
      const innerContainer = fluidContainer["container"] as Container;
      innerContainer.attach(request);
  }
```

For other services, such as ODSP, I was thinking there could being an extension of the FluidContainer to M365FluidContainer to add capabilities like supporting sharing links. This could include the generic definition of the ContainerConfig as part of the service-specific class definition.

Reply:
This type of thing is a common issue when wrapping. Personally, i would provide an escape hatch from the wrapper by exposing container